providers/implementations/storemgmt/file_store.c

```diff
@@ -190,7 +190,7 @@ static void *file_open_dir(const char *path, const char *uri, void *provctx)
     }
     return ctx;
  err:
-    file_close(ctx);
+    free_file_ctx(ctx);
```

Comment:
Is this really the correct behaviour? We can end up here if `OPENSSL_DIR_read` returns a NULL value and errno != 0. If that occurs but `OPENSSL_DIR_read` still populated `ctx->_.dir.ctx` then `file_close` still seems like the correct thing to do. Looking at LPdir_unix.c it seems that this could actually happen if the readdir call here returns NULL:

https://github.com/openssl/openssl/blob/dd2fcc1f7c44c5fb5aa2d33aecdc699c7018ce01/crypto/LPdir_unix.c#L129-L132

Reply:
Ah, you're right.

Reply:
Return file_close back?

Reply:
So the correct fix is to just return NULL instead of the first `goto err;` in the function.

Reply:
No problem. Need some time. Do my way home.